Sources/SwiftSyntaxMacroExpansion/MacroExpansion.swift

```diff
@@ -215,7 +215,7 @@ public func expandAttachedMacroWithoutCollapsing<Context: MacroExpansionContext>
   definition: Macro.Type,
   macroRole: MacroRole,
   attributeNode: AttributeSyntax,
-  declarationNode: DeclSyntax,
+  declarationNode: some SyntaxProtocol,
```

Comment:
`declarationNode` is no longer the correct argument label here if this can be a closure as well, right?

Reply:
That's correct, ideally the argument label should be called `attachedTo` or just `node` or similar. However, I was trying to maintain source compatibility since this is a public function, per @rintaro 's recommendation, though I'm not sure what kind of code actually calls `expandAttachedMacro{WithoutCollapsing}` from outside of swift-syntax. I'm happy to change the argument label in both APIs if you aren't concerned about source compatibility.

Reply:
I think we should change it and have a deprecated version of this function with the old signature that forwards to the new version.

Reply:
Done!